src/gt4py/stencil_object.py

```diff
@@ -35,6 +36,15 @@
 OriginType = Union[Tuple[int, int, int], Dict[str, Tuple[int, ...]]]
 
 
+def _compute_cache_key(field_args, parameter_args, domain, origin) -> int:
+    field_data = tuple(
+        (name, field.shape, field.default_origin)
+        for name, field in field_args.items()
+        if field is not None
+    )
+    return hash((*field_data, *parameter_args.keys(), dumps(domain), dumps(origin)))
```

Comment:
No need to unpack `field_data` tuple since it is hashable.
```suggestion
    return hash((field_data, *parameter_args.keys(), dumps(domain), dumps(origin)))
```